ec/src/models/bn/mod.rs

```diff
@@ -18,10 +18,16 @@ pub enum TwistType {
 }
 
 pub trait BnParameters: 'static {
+    // The absolute value of the BN curve parameter `X` (as in `q = 36 X^4 + 36 X^3 + 24 X^2 + 6 X + 1`).
     const X: &'static [u64];
+    // Whether or not `X` is negative.
     const X_IS_NEGATIVE: bool;
+
+    // The absolute value of `6X + 2`.
     const ATE_LOOP_COUNT: &'static [i8];
+    // Whether or not `6X + 2` is negative. `6X + 2` is negative often when `X` is negative.
```

Comment:
Isn't 6X + 2 negative *exactly* when X is negative? (Because X is an integer and 2 < 6.) That implies that this field should probably just be removed and `X_IS_NEGATIVE` used instead.

Reply:
Also I'm confused because this doc appears to contradict the following sentence in the PR description:
> This PR is to clarify that ATE_LOOP_COUNT is 6 X + 2 instead of |6 X + 2|, a possible confusion.

Reply:
You are right. 

Let me clarify this PR's documentation as well as the documentation in the code (you are right, 2 < 6 so X must be negative). 

I think there are two possible future plans:

1. Not too breaking, which can be done now:
- Use `X_IS_NEGATIVE ` to decide whatever `ATE_LOOP_COUNT_IS_NEGATIVE` is deciding, since they should be equivalent for BN curves.

2. Breaking:
- Remove `ATE_LOOP_COUNT_IS_NEGATIVE` and update the BN254 implementation accordingly.